pkg/data/components/RedactionProcessor.yaml

```diff
@@ -0,0 +1,65 @@
+kind: RedactionProcessor
+name: Redaction Processor
+style: processor
+type: base
+status: development
+version: v0.0.1
+summary: A processor that can be used to redact attribute values in telemetry.
+description: |
+  This processor allows you to redact attribute values from traces, metrics, and logs based on
+  specified rules.
+tags:
+  - category:processor
+  - service:collector
+  - category:filter
+  - signal:OTelTraces
+  - signal:OTelMetrics
+  - signal:OTelLogs
+ports:
+  - name: Traces
+    direction: input
+    type: OTelTraces
+  - name: Metrics
+    direction: input
+    type: OTelMetrics
+  - name: Logs
+    direction: input
+    type: OTelLogs
+properties:
+  - name: AttributeNamePatterns
+    summary: Regular expression patterns of attribute names to redact.
+    description: |
+      A list of regular expression patterns that match attribute names to be redacted.
+      If an attribute name matches any of these patterns, its value will be redacted.
+    type: stringarray
+    validations:
+      - noblanks
+      - nonempty
+    default: []
+  - name: AttributeValuePatterns
+    summary: Regular expression patterns of attribute values to redact.
+    description: |
+      A list of regular expression patterns that match attribute values to be redacted.
+      If an attribute value matches any of these patterns, it will be redacted.
+    type: stringarray
+    validations:
+      - noblanks
+      - nonempty
+    default: []
```

Comment:
I think we should keep the defaults from the redaction processor (card numbers). It's easy to delete them, and gives a good user experience that shows what it might look like and that they can add the component without changes.